### DISCUSSION TITLE
How to check that a lambda layer has been added before adding it

### DISCUSSION THREAD

**Discussion by jabalsad** [aws/aws-cdk#24040] [General]
If I read the source code correctly, there is no way to check whether a lambda function layer already exists before adding it.

The `lambda.Function` Construct only allows you to either:
- supply a list of layers to add as part of the construct props `lambda.FunctionProps`
- call the `fn.addLayers(...)` function defined [here](https://github.com/aws/aws-cdk/blob/v2.63.1/packages/@aws-cdk/aws-lambda/lib/function.ts#L976)

How would I check to see if a layer already exists on a function before adding it (after a function construct has already been created)? It doesn't seem like I should be accessing `_layers` variable directly.

**Comment by indrora** [comment]
Since the CDK is a wrapper around CloudFormation, I don't know that this is possible until CloudFormation deployment time.
